hnn_core/network.py

```diff
@@ -14,6 +14,54 @@
 from .params import create_pext
 
 
+def read_spikes(fname, gid_dict=None):
+    """Read spiking activity from a collection of spike trial
+    files.
+
+    Parameters
+    ----------
+    fname : str
+        Wildcard expression (e.g., '<pathname>/spk_*.txt') of the
+        path to the spike file(s).
+    gid_dict : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, each spike file must contain
+        a 3rd column for spike type.
+
+    Returns
+    ----------
+    spikes : Spikes object
+        An instance of the Spikes object.
+    """
+    from glob import glob
+
+    spiketimes = ()
+    spikegids = ()
+    spiketypes = ()
+    for file in sorted(glob(fname)):
+        spike_trial = np.loadtxt(file, dtype=str)
+        spiketimes += (list(spike_trial[:, 0].astype(float)),)
+        spikegids += (list(spike_trial[:, 1].astype(int)),)
+
+        # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
+        # spike type. If reading a legacy version, validate that a gid_dict is
+        # provided.
+        if spike_trial.shape[1] == 3:
+            spiketypes += (list(spike_trial[:, 2].astype(str)),)
+        else:
+            assert gid_dict is not None, ("Error: gid_dict must be provided "
+                                          "if spike types are unspecified in "
+                                          "'spk.txt' file")
```

Comment:
The right way to raise errors is:

```suggestion
            if gid_dict is None:
                raise ValueError('gid_dict must be provided blah blah')
```

then you should add a test to check if these errors are actually being tested You can use `pytest.raises`